src/runtime/tests/manifest-test.ts

```diff
@@ -3659,6 +3659,130 @@ resource SomeName
       `), `Type variable ~a does not contain field 'foo'`);
     });
 
+    it('fails to parse concrete types with inline schema field of `*`', async () => {
+      await assertThrowsAsync(async () => await parseManifest(`
+          particle Foo
+            data: reads {*}
+      `), `\
+Post-parse processing error caused by 'undefined' line 3.
+Only type variables may have '*' fields.
+              data: reads {*}
+                          ^^^`);
+      await assertThrowsAsync(async () => await parseManifest(`
+          particle Foo
+            data: reads {name: Text, *}
+      `), `\
+Post-parse processing error caused by 'undefined' line 3.
+Only type variables may have '*' fields.
+              data: reads {name: Text, *}
+                          ^^^^^^^^^^^^^^^`);
+    });
+
+    it('warns about using multiple `*` in a single variable constraint', async () => {
+      const manifest = await parseManifest(`
+          particle Foo
+            data: reads ~a with {*, *}
+      `);
+
+      assert.lengthOf(manifest.errors, 1);
+      assert.equal(manifest.errors[0].key, 'multiStarFields');
+    });
+
+    it('supports field-level checks and claims with max type variables', async () => {
+      const manifest = await parseManifest(`
+        particle OrderIngestion in '.OrderIngestion'
+          data: writes [Product {sku: Text, name: Text, price: Number}]
+
+        particle SkuRedactor in '.SkuRedactor'
+          input: reads [~a with {sku: Text}]
+          output: writes [~a]
+          claim output.sku is redacted
```

Comment:
Now that you are testing "Supports field-level checks and claims with max type variables" why are you still having such a large recipe and doing all that stuff?

it should be enough to have a particle like that:
```
        particle SkuRedactor in '.SkuRedactor'
          input: reads [~a with {sku: Text}]
          output: writes [~a]
          check input.sku is great
          claim output.sku is awesome
```

parse it, and then ensure checks and claims are placed on the particle spec. You are not running DFA below, also DFA for typevars is not yet done. There is nothing more you can do at this stage.